include/nbl/ui/CWindowWin32.h

```diff
@@ -39,97 +41,150 @@ class CWindowWin32 final : public IWindowWin32
 	}
 
 private:
-    CWindowWin32(core::smart_refctd_ptr<system::ISystem>&& sys, uint32_t _w, uint32_t _h, E_CREATE_FLAGS _flags) : IWindowWin32(std::move(sys), _w, _h, _flags), m_native(NULL)
+    CWindowWin32(core::smart_refctd_ptr<system::ISystem>&& sys, uint32_t _w, uint32_t _h, E_CREATE_FLAGS _flags) : 
+		IWindowWin32(std::move(sys), _w, _h, _flags), m_threadHandler(_w, _h, _flags)
     {
-		// get process handle
-		HINSTANCE hinstance = GetModuleHandle(NULL);
-
-		const char* classname = __TEXT("Nabla Engine");
-
-		WNDCLASSEX wcex;
-		wcex.cbSize = sizeof(WNDCLASSEX);
-		wcex.style = CS_HREDRAW | CS_VREDRAW;
-		wcex.lpfnWndProc = WndProc;
-		wcex.cbClsExtra = 0;
-		wcex.cbWndExtra = 0;
-		wcex.hInstance = hinstance;
-		wcex.hIcon = NULL;
-		wcex.hCursor = 0; // LoadCursor(NULL, IDC_ARROW);
-		wcex.hbrBackground = (HBRUSH)(COLOR_WINDOW + 1);
-		wcex.lpszMenuName = 0;
-		wcex.lpszClassName = classname;
-		wcex.hIconSm = 0;
-
-		RegisterClassEx(&wcex);
-
-		// calculate client size
-
-		RECT clientSize;
-		clientSize.top = 0;
-		clientSize.left = 0;
-		clientSize.right = m_width;
-		clientSize.bottom = m_height;
-
-		DWORD style = WS_POPUP; // TODO why popup?
-
-		if (!isFullscreen())
+		m_width = _w; m_height = _h;
+		SetWindowLongPtr(m_threadHandler.getNativeWindow(), GWLP_USERDATA, reinterpret_cast<LONG_PTR>(this));
+    }
+
+
+    //native_handle_t m_native;
+private:
+	class CThreadHandler final : public system::IThreadHandler<CThreadHandler>
+	{
+		using base_t = system::IThreadHandler<CThreadHandler>;
+		friend base_t;
+	public:
+		CThreadHandler(uint32_t _w, uint32_t _h, E_CREATE_FLAGS _flags)
 		{
-			if (!isBorderless())
+			HINSTANCE hinstance = GetModuleHandle(NULL);
+
+			const char* classname = __TEXT("Nabla Engine");
```

Comment:
TODO: add a creation parameter for the window title, and add a method to `IWindow` for changing the window title at runtime.

This also needs a capability query (because not every platform has window titles or even if they do they might not be changeable).

Reply:
eh I just noticed this is the class name... sorry, resolving.